webxr/resources/webxr_util.js

```diff
@@ -161,7 +163,7 @@ function forEachWebxrObject(callback) {
 
 // Code for loading test API in Chromium.
 let loadChromiumResources = Promise.resolve().then(() => {
```

Comment:
While this approach looks good to me; there is still a bit of a potential for bugs. I hit one with the "isWebKitBased" check not being "good enough" in setupWebkitWebXRTestAPI, since they were different checks, and as a variable promise, it was being run no matter what. And obviously you all hit a similar issue (hence this fix). It feels a bit odd that we need to ensure that both the awaits and the promises need wrap the same bools.

I wonder if it would be better to convert the two promises (loadChromiumResources/setupWebKitWebXRTestAPI) to functions that return a promise so that we don't need to check twice?

Reply:
Yeah, I also wondered why they were defined as variable promises instead of functions that return promises. I'll change that.

Reply:
Oh wait; I think we do still want *part* of this in place.  One thing that the defined promise (and the window.XRTest check) did for us was ensure that they only ran once, even if there were multiple xr_promise_tests (or things that called xr_promise_tests) in a single file.

Maybe consider a wrapper like:
```
// Check if the test API needs to be polyfilled in
if (!navigator.xr.test) {
  if (isChromiumBased) {
...
 } else if (isWebKitBased) {
... 
} else {
  // Friendly failure message that navigator.xr.test is undefined and there is no known "polyfill"?
}
}